### PR TITLE
[#159241621]Fix delegate, set initial slider value

### PR DESCRIPTION
**Pull request by ivan-bui**

### CODE REVIEW

File: VouchrSdkShowCase-iOS/Collection View Cells/SliderCollectionViewCell.m [hunk 1]
```diff
@@ -30,6 +30,7 @@ - (void)setupWithDescription:(NSString *)description
     self.sliderView = [SliderView new];
     [self.sliderView setupWithPropertyName:description minValue:minValue maxValue:maxValue delegate:delegate];
     self.sliderView.translatesAutoresizingMaskIntoConstraints = NO;
+    [self.sliderView setCurrentValue:(maxValue + minValue) / 2];
```
Comment (rbui) [inline VouchrSdkShowCase-iOS/Collection View Cells/SliderCollectionViewCell.m]: i deleted this on purpose.
This should not be done. 
Remove this

Comment (rbui) [reply VouchrSdkShowCase-iOS/Collection View Cells/SliderCollectionViewCell.m]: Theres a method to set the value which should be based on the value from the personalization option

Comment (rbui) [reply VouchrSdkShowCase-iOS/Collection View Cells/SliderCollectionViewCell.m]: prob need to add a method to the cell to be able to update the slider value, we want it based on the personalization property value, not a random one

Comment (ivan-bui) [reply VouchrSdkShowCase-iOS/Collection View Cells/SliderCollectionViewCell.m]: huh? are there values in personalization properties? probably need to blow up the collectionView at this point.

Comment (rbui) [reply VouchrSdkShowCase-iOS/Collection View Cells/SliderCollectionViewCell.m]: yeah, prob.
Explained in ur previous PR. If there personalizationOption was instantiated with the builder, regardless if properties are set, there should be default values.

**Comment by ivan-bui** [comment]
this beauty was designed by @alicehuanggg 

screen shot 2018-07-25 at 12 21 13 pm
screen shot 2018-07-25 at 12 45 57 pm

**Comment by ivan-bui** [comment]
designed by @alicehuanggg 

screen shot 2018-07-25 at 4 20 59 pm
screen shot 2018-07-25 at 4 21 11 pm

**Comment by alicehuanggg** [comment]
ew can you fix

File: VouchrSdkShowCase-iOS.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -1672,11 +1563,7 @@
 				CODE_SIGN_STYLE = Automatic;
 				DEVELOPMENT_TEAM = XBS3LC7V64;
 				INFOPLIST_FILE = "VouchrSdkShowCase-iOSUITests/Info.plist";
-				LD_RUNPATH_SEARCH_PATHS = (
-					"$(inherited)",
-					"@executable_path/Frameworks",
-					"@loader_path/Frameworks",
-				);
```
Comment (rbui) [inline VouchrSdkShowCase-iOS.xcodeproj/project.pbxproj]: dont check this file in unless the changes are relevant to your work. 
Its supposed to be ignored by git, i gotta fix it.

File: VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/VideoCustomizationCollectionViewCell.m [hunk 3]
```diff
@@ -52,29 +61,40 @@ - (UICollectionViewLayoutAttributes *)preferredLayoutAttributesFittingAttributes
...
-    self.boolProperties = [config.personalizationOption boolPropertyNames];
-    NSMutableArray *nonBoolPropertyNames = [[config.personalizationOption propertyNames] mutableCopy];
-    [nonBoolPropertyNames removeObjectsInArray:[config.personalizationOption boolPropertyNames]];
-    self.nonBoolProperties = nonBoolPropertyNames;
     self.titleLabel.text = config.name;
+    [self setupConfigurationControls];
+}
+
+- (void)setupConfigurationControls {
+    [self.canTakeVideoCheckboxView setupWithDescription:@"Take Video" isChecked:NO delegate:self];
```
Comment (rbui) [inline VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/VideoCustomizationCollectionViewCell.m]: Will this have the same issue as before when reused? 
isChecked value not based on model

File: VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/VideoCustomizationCollectionViewCell.m [hunk 4]
```diff
@@ -52,29 +61,40 @@ - (UICollectionViewLayoutAttributes *)preferredLayoutAttributesFittingAttributes
...
+    [self.maxVideoLengthSliderView setupWithPropertyName:@"maxVideoLength" minValue:0 maxValue:500 delegate:self];
+    [self.maxVideoLengthSliderView setCurrentValue:0];
```
Comment (rbui) [inline VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/VideoCustomizationCollectionViewCell.m]: shouldnt these current values be based on the personalization otpion

File: VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/VideoCustomizationCollectionViewCell.m [hunk 5]
```diff
@@ -52,29 +61,40 @@ - (UICollectionViewLayoutAttributes *)preferredLayoutAttributesFittingAttributes
...
+    
+    [self.canAddYoutubeCheckboxView setupWithDescription:@"Add Youtube" isChecked:YES delegate:self];
+    [self.canSearchYoutubeCheckboxView setupWithDescription:@"Search Youtube" isChecked:YES delegate:self];
+    [self.canClipYoutubeCheckboxView setupWithDescription:@"Can Clip Youtube" isChecked:YES delegate:self];
+    [self.maxYoutubeSearchResultSliderView setupWithPropertyName:@"maxYoutubeSearchResults" minValue:0 maxValue:250 delegate:self];
+    [self.maxYoutubeSearchResultSliderView setCurrentValue:50];
+
+    [self.config.personalizationOption setValue:@(YES) forKey:@"canAddYoutubeVideos"];
+    [self.config.personalizationOption setValue:@(YES) forKey:@"canSearchYoutube"];
+    [self.config.personalizationOption setValue:@(YES) forKey:@"canClipYoutube"];
```
Comment (rbui) [inline VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/VideoCustomizationCollectionViewCell.m]: same goes for everything in this method.

File: VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/VideoCustomizationCollectionViewCell.m [hunk 6]
```diff
@@ -95,54 +115,62 @@ - (IBAction)checkboxButtonTapped:(id)sender {
...
+    }
     [self.delegate customizationCellUpdatedPersonalization:self.config];
 }
 
 - (void)checkboxViewUnselected:(CheckboxView *)checkboxView {
-    [self.config.personalizationOption setValue:@(NO) forKey:checkboxView.descriptionLabelText];
-    [self.delegate customizationCellUnselectedPersonalization:self.config];
+    if (checkboxView == self.canAddYoutubeCheckboxView) {
+        [self.config.personalizationOption setValue:@(NO) forKey:@"canAddYoutubeVideos"];
+        self.canAddYoutubeOverlayView.alpha = 0.52f;
```
Comment (rbui) [inline VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/VideoCustomizationCollectionViewCell.m]: whats with these weird alpha values? the 0.02 isn't noticeable. 
also, why isn't it a constant or a method.

File: VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/VideoCustomizationCollectionViewCell.m [hunk 7]
```diff
@@ -94,55 +112,78 @@ - (IBAction)checkboxButtonTapped:(id)sender {
...
+}
 
-- (NSInteger)collectionView:(UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section {
-    if (collectionView == self.boolCollectionView) {
-        return self.boolProperties.count;
-    } else {
-        return self.nonBoolProperties.count;
-    }
+- (void)disableTakeVideoPersonalizationOptions {
+    self.canTakeVideoOverlayView.alpha = 0.5f;
```
[the remark this answers is not in the document]
Comment (ivan-bui) [reply VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/VideoCustomizationCollectionViewCell.m]: whats that?

[the remark this answers is not in the document]
Comment (rbui) [reply VouchrSdkShowCase-iOS/Personalization Section/Collection View Cells/VideoCustomizationCollectionViewCell.m]: we talked about it 5mins ago

**Pull request merged**